src/MassTransit/Pipeline/Pipes/ResponsePipe.cs

```diff
@@ -36,6 +36,13 @@ public Task Send(SendContext context)
             context.RequestId = _context.RequestId;
             context.SourceAddress = _context.ReceiveContext.InputAddress;
 
+            if (_context.ExpirationTime.HasValue)
+            {
+                context.TimeToLive = _context.ExpirationTime.Value - DateTime.UtcNow;
+                if (context.TimeToLive.Value <= TimeSpan.Zero)
+                    context.TimeToLive = TimeSpan.FromSeconds(1);

```

Comment:
maybe be better to have 5sec for example?

Reply:
If the originator has already timed out, it's pretty pointless, so this would mean the message would be delivered if the consumer is waiting for it. Otherwise, it disappears.